proc.go

```diff
@@ -279,3 +279,21 @@ func (p Proc) fileDescriptors() ([]string, error) {
 func (p Proc) path(pa ...string) string {
 	return p.fs.Path(append([]string{strconv.Itoa(p.PID)}, pa...)...)
 }
+
+// FileDescriptorsInfo retrieves information about all file descriptors of
+// the process.
+func (p Proc) FileDescriptorsInfo() (ProcFDInfos, error) {
+	names, err := p.fileDescriptors()
+	if err != nil {
+		return nil, err
+	}
+
+	fdinfos := make(ProcFDInfos, len(names))
+
+	for i, n := range names {
+		fdinfo, _ := p.FDInfo(n)
+		fdinfos[i] = *fdinfo
```

Comment:
If `p.FDInfo` returns an error and a nil fdinfo won't this cause a nil pointer dereference?  It seems like this could happen for example if a file is closed while iterating through the files.  I think it would be better to check for an error and then use `append` to add to the slice instead of using the index.

Also, if you make `ProcFDInfos` a slice of pointers instead of a slice of structs, then it should be a bit more efficient because the full struct don't need to be copied with each assignment.